Agent crashes with spelling mistake on exported function

### What's wrong?

Got the error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x580022b]
goroutine 1 [running]:
github.com/grafana/agent/component/otelcol/internal/fanoutconsumer.Logs({0xc002d41a60, 0x2, 0x0?})
 /src/agent/component/otelcol/internal/fanoutconsumer/logs.go:43 +0x2ab
github.com/grafana/agent/component/otelcol/processor.(*Processor).Update(0xc002359ad0, {0x7069f00?, 0xc002d68630})
 /src/agent/component/otelcol/processor/processor.go:160 +0x5c8
github.com/grafana/agent/component/otelcol/processor.New({{0x7f23b84c21b0, 0xc002a50030}, {0xc00266d110, 0x24}, {0x86b0340, 0xc002a43860}, {0xc00266d140, 0x2f}, 0xc002a38ee0, {0x86ff2e0, ...}, ...}, ...)
 /src/agent/component/otelcol/processor/processor.go:100 +0x345
github.com/grafana/agent/component/otelcol/processor/attributes.init.0.func1({{0x7f23b84c21b0, 0xc002a50030}, {0xc00266d110, 0x24}, {0x86b0340, 0xc002a43860}, {0xc00266d140, 0x2f}, 0xc002a38ee0, {0x86ff2e0, ...}, ...}, ...)
 /src/agent/component/otelcol/processor/attributes/attributes.go:26 +0xf2
github.com/grafana/agent/pkg/flow/internal/controller.(*ComponentNode).evaluate(0xc002a4b8c0, 0x0?)
 /src/agent/pkg/flow/internal/controller/node_component.go:279 +0x459
github.com/grafana/agent/pkg/flow/internal/controller.(*ComponentNode).Evaluate(0xc002636820?, 0xcd98a00?)
 /src/agent/pkg/flow/internal/controller/node_component.go:251 +0x1c
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).evaluate(0xc001c3d680, {0x86b0340, 0xc002a640f0}, {0x8709b30?, 0xc002a4b8c0?})
 /src/agent/pkg/flow/internal/controller/loader.go:701 +0x58
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).Apply.func2({0x86b3ea0, 0xc002a4b8c0})
 /src/agent/pkg/flow/internal/controller/loader.go:177 +0x585
github.com/grafana/agent/pkg/flow/internal/dag.WalkTopological(0xc002a3d5c0, {0xc002a44200, 0xe, 0x2?}, 0xc001de7058)
 /src/agent/pkg/flow/internal/dag/walk.go:83 +0x226
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).Apply(0xc001c3d680, 0xf80adc6659151211?, {0xc002a44000?, 0x19, 0x20}, {0xc00118ed08, 0x1, 0x1})
 /src/agent/pkg/flow/internal/controller/loader.go:160 +0xaf8
github.com/grafana/agent/pkg/flow.(*Flow).LoadSource(0xc0021a0a50, 0xc002a36a80, 0x764da68?)
 /src/agent/pkg/flow/flow.go:283 +0xd6
github.com/grafana/agent/cmd/internal/flowmode.(*flowRun).Run.func4()
 /src/agent/cmd/internal/flowmode/cmd_run.go:269 +0x268
github.com/grafana/agent/cmd/internal/flowmode.(*flowRun).Run(0xc001048a80, {0x7ffd146ac560, 0x17})
 /src/agent/cmd/internal/flowmode/cmd_run.go:302 +0xde2
github.com/grafana/agent/cmd/internal/flowmode.runCommand.func1(0xc001023100?, {0xc002631860?, 0x4?, 0x764da9c?})
 /src/agent/cmd/internal/flowmode/cmd_run.go:92 +0x2b
github.com/spf13/cobra.(*Command).execute(0xc00263c300, {0xc0026317d0, 0x3, 0x3})
 /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc001c57800)
 /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
 /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/grafana/agent/cmd/internal/flowmode.Run()
 /src/agent/cmd/internal/flowmode/flowmode.go:33 +0x305
main.main()
 /src/agent/cmd/grafana-agent/main.go:50 +0x6bd
```


### Steps to reproduce

All I did was spell "receiver"incorrectly as "reciever"

### System information

Linux (Kubernetes) Based on debian 12.

### Software version

0.37.3

### Configuration

```text
logging {
      level  = "info"
      format = "logfmt"
    }

    discovery.kubernetes "pods" {
      role = "pod"
    }

    discovery.kubernetes "nodes" {
      role = "node"
    }

    discovery.kubernetes "services" {
      role = "service"
    }

    discovery.kubernetes "endpoints" {
      role = "endpoints"
    }

    discovery.kubernetes "endpointslices" {
      role = "endpointslice"
    }

    discovery.kubernetes "ingresses" {
      role = "ingress"
    }

    prometheus.scrape "openweathermap_scraper" {
      targets = [
        {"__address__" = "weather-exporter1.otel-demo:2112"},
      ]

      forward_to = [
        prometheus.remote_write.enterprise_metrics.receiver,
        prometheus.remote_write.mimir_distributed.receiver,
        otelcol.receiver.prometheus.default.receiver,
      ]

      scrape_interval = "10s"
      metrics_path    = "/metrics"
    }

    prometheus.scrape "coredns_scraper" {
      targets = [
        {"__address__" = "coredns-coredns.kube-system.svc:9153", "job" = "coredns"},
        {"__address__" = "coredns-coredns.kube-system.svc:9154", "job" = "lab.home"},
      ]

      forward_to = [
        prometheus.remote_write.enterprise_metrics.receiver,
        prometheus.remote_write.mimir_distributed.receiver,
        otelcol.receiver.prometheus.default.receiver,
      ]

      scrape_interval = "10s"
      metrics_path    = "/metrics"
    }

    prometheus.scrape "tns_cloud_scraper" {
      targets = [
        {"__address__" = "app.tns-cloud.svc"},
        {"__address__" = "db.tns-cloud.svc"},
        {"__address__" = "loadgen.tns-cloud.svc"},
      ]

      forward_to = [
        prometheus.remote_write.enterprise_metrics.receiver,
        prometheus.remote_write.mimir_distributed.receiver,
        otelcol.receiver.prometheus.default.receiver,
      ]

      scrape_interval = "10s"
      metrics_path    = "/metrics"
    }

    prometheus.scrape "cilium_scraper" {
      targets = [
        {"__address__" = "hubble-metrics.kube-system:9965"},
      ]

      forward_to = [
        prometheus.remote_write.enterprise_metrics.receiver,
        prometheus.remote_write.mimir_distributed.receiver,
        otelcol.receiver.prometheus.default.receiver,
      ]

      scrape_interval = "10s"
      metrics_path    = "/metrics"
    }

    prometheus.scrape "cert_manager_scraper" {
      targets = [
        {"__address__" = "trust-manager-metrics.cert-manager:9402"},
        {"__address__" = "cert-manager.cert-manager:9402"},
      ]

      forward_to = [
        prometheus.remote_write.enterprise_metrics.receiver,
        prometheus.remote_write.mimir_distributed.receiver,
        otelcol.receiver.prometheus.default.receiver,
      ]

      scrape_interval = "10s"
      metrics_path    = "/metrics"
    }

    

    otelcol.receiver.jaeger "jaeger" {
      protocols {
        grpc {}
        thrift_http {}
        thrift_binary {}
        thrift_compact {}
      }

      output {
        metrics = [otelcol.processor.attributes.default.input]
        logs    = [otelcol.processor.attributes.default.input]
        traces  = [otelcol.processor.attributes.default.input]
      }
    }

    otelcol.auth.basic "creds" {
      username = "485303"
    }

    otelcol.auth.basic "tempo" {
      username = "485303"
    }

    otelcol.auth.basic "enterprise_metrics" {
      username = "default"
    }

    otelcol.auth.basic "enterprise_traces" {
      username = "default"
    }

    otelcol.receiver.prometheus "default" {
      output {
        metrics = [
          otelcol.processor.attributes.default.input,
        ]
      }
    }

    otelcol.processor.attributes "default" {
      action {
          key = "cluster"
          value = "kubernetes"
          action = "insert"
      }

      output {
        metrics = [
          otelcol.exporter.otlphttp.grafana_cloud.input,
        ]
        logs    = [
          otelcol.exporter.otlphttp.grafana_cloud.input,
          otelcol.exporter.loki.default.input,
        ]
        traces  = [
          otelcol.exporter.otlphttp.grafana_cloud.input,
          otelcol.exporter.otlp.enterprise_traces.input,
        ]
      }
    }

    otelcol.exporter.loki "default" {
      forward_to = [
        loki.write.enterprise_logs.receiver,
        loki.write.loki_distributed.reciever,
      ]
    }

    otelcol.exporter.otlphttp "grafana_cloud" {
        client {
            endpoint = "https://otlp-gateway-prod-us-central-0.grafana.net/otlp"

            tls {
                insecure             = false
                insecure_skip_verify = false
            }

            auth = otelcol.auth.basic.creds.handler
        }
    }

    otelcol.exporter.otlp "enterprise_traces" {
        client {
            endpoint = "get1-tempo-enterprise-gateway.enterprise-traces:3100"

            tls {
                insecure             = true
                insecure_skip_verify = true
            }
            
            auth = otelcol.auth.basic.enterprise_traces.handler
        }
    }

    prometheus.remote_write "enterprise_metrics" {
        endpoint {
            url = "http://gem1-mimir-gateway.enterprise-metrics/api/v1/push"

            basic_auth {
                username = "default"
            }
        }

        external_labels = {
            cluster = "kubernetes",
        }
    }

    prometheus.remote_write "mimir_distributed" {
        endpoint {
            url = "http://mimir-nginx.mimir/api/v1/push"
        }

        external_labels = {
            cluster = "kubernetes",
        }
    }

    loki.write "loki_distributed" {
      endpoint {
        url = "http://loki-loki-distributed-gateway.loki/loki/api/v1/push"
      }
      external_labels = {
        cluster = "kubernetes",
      }
    }

    loki.write "enterprise_logs" {
      endpoint {
        url = "http://gel1-enterprise-logs-gateway.enterprise-logs:3100/loki/api/v1/push"

        basic_auth {
          username = "default"
        }
      }
      external_labels = {
        cluster = "kubernetes",
      }
    }
```
```


### Logs

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x580022b]
goroutine 1 [running]:
github.com/grafana/agent/component/otelcol/internal/fanoutconsumer.Logs({0xc002d41a60, 0x2, 0x0?})
 /src/agent/component/otelcol/internal/fanoutconsumer/logs.go:43 +0x2ab
github.com/grafana/agent/component/otelcol/processor.(*Processor).Update(0xc002359ad0, {0x7069f00?, 0xc002d68630})
 /src/agent/component/otelcol/processor/processor.go:160 +0x5c8
github.com/grafana/agent/component/otelcol/processor.New({{0x7f23b84c21b0, 0xc002a50030}, {0xc00266d110, 0x24}, {0x86b0340, 0xc002a43860}, {0xc00266d140, 0x2f}, 0xc002a38ee0, {0x86ff2e0, ...}, ...}, ...)
 /src/agent/component/otelcol/processor/processor.go:100 +0x345
github.com/grafana/agent/component/otelcol/processor/attributes.init.0.func1({{0x7f23b84c21b0, 0xc002a50030}, {0xc00266d110, 0x24}, {0x86b0340, 0xc002a43860}, {0xc00266d140, 0x2f}, 0xc002a38ee0, {0x86ff2e0, ...}, ...}, ...)
 /src/agent/component/otelcol/processor/attributes/attributes.go:26 +0xf2
github.com/grafana/agent/pkg/flow/internal/controller.(*ComponentNode).evaluate(0xc002a4b8c0, 0x0?)
 /src/agent/pkg/flow/internal/controller/node_component.go:279 +0x459
github.com/grafana/agent/pkg/flow/internal/controller.(*ComponentNode).Evaluate(0xc002636820?, 0xcd98a00?)
 /src/agent/pkg/flow/internal/controller/node_component.go:251 +0x1c
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).evaluate(0xc001c3d680, {0x86b0340, 0xc002a640f0}, {0x8709b30?, 0xc002a4b8c0?})
 /src/agent/pkg/flow/internal/controller/loader.go:701 +0x58
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).Apply.func2({0x86b3ea0, 0xc002a4b8c0})
 /src/agent/pkg/flow/internal/controller/loader.go:177 +0x585
github.com/grafana/agent/pkg/flow/internal/dag.WalkTopological(0xc002a3d5c0, {0xc002a44200, 0xe, 0x2?}, 0xc001de7058)
 /src/agent/pkg/flow/internal/dag/walk.go:83 +0x226
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).Apply(0xc001c3d680, 0xf80adc6659151211?, {0xc002a44000?, 0x19, 0x20}, {0xc00118ed08, 0x1, 0x1})
 /src/agent/pkg/flow/internal/controller/loader.go:160 +0xaf8
github.com/grafana/agent/pkg/flow.(*Flow).LoadSource(0xc0021a0a50, 0xc002a36a80, 0x764da68?)
 /src/agent/pkg/flow/flow.go:283 +0xd6
github.com/grafana/agent/cmd/internal/flowmode.(*flowRun).Run.func4()
 /src/agent/cmd/internal/flowmode/cmd_run.go:269 +0x268
github.com/grafana/agent/cmd/internal/flowmode.(*flowRun).Run(0xc001048a80, {0x7ffd146ac560, 0x17})
 /src/agent/cmd/internal/flowmode/cmd_run.go:302 +0xde2
github.com/grafana/agent/cmd/internal/flowmode.runCommand.func1(0xc001023100?, {0xc002631860?, 0x4?, 0x764da9c?})
 /src/agent/cmd/internal/flowmode/cmd_run.go:92 +0x2b
github.com/spf13/cobra.(*Command).execute(0xc00263c300, {0xc0026317d0, 0x3, 0x3})
 /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc001c57800)
 /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
 /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/grafana/agent/cmd/internal/flowmode.Run()
 /src/agent/cmd/internal/flowmode/flowmode.go:33 +0x305
main.main()
 /src/agent/cmd/grafana-agent/main.go:50 +0x6bd
```
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -39,6 +39,8 @@ Main (unreleased)
 
 - Fix issue with windows_exporter defaults not being set correctly. (@mattdurham)
 
+- Fix agent crash when process null OTel's fan out consumers. (@hainenber)
+
 v0.38.0 (2023-11-21)
 --------------------
 

--- a/component/otelcol/internal/fanoutconsumer/logs.go
+++ b/component/otelcol/internal/fanoutconsumer/logs.go
@@ -29,6 +29,10 @@ func Logs(in []otelcol.Consumer) otelconsumer.Logs {
 	for i := 0; i < len(in)-1; i++ {
 		consumer := in[i]
 
+		if consumer == nil {
+			continue
+		}
+
 		if consumer.Capabilities().MutatesData {
 			clone = append(clone, consumer)
 		} else {
@@ -40,10 +44,12 @@ func Logs(in []otelcol.Consumer) otelconsumer.Logs {
 
 	// The final consumer can be given to the passthrough list regardless of
 	// whether it mutates as long as there's no other read-only consumers.
-	if len(passthrough) == 0 || !last.Capabilities().MutatesData {
-		passthrough = append(passthrough, last)
-	} else {
-		clone = append(clone, last)
+	if last != nil {
+		if len(passthrough) == 0 || !last.Capabilities().MutatesData {
+			passthrough = append(passthrough, last)
+		} else {
+			clone = append(clone, last)
+		}
 	}
 
 	return &logsFanout{

